common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -120,14 +120,14 @@ export class ApplicationTokenCredentials {
         return this.clientId;
     }
 
-    private _getMSIAuthorizationToken(retyCount: number ,timeToWait: number): Q.Promise<string> {
+    public static getMSIAuthorizationToken(retyCount: number ,timeToWait: number, baseUrl: string, msiClientId?: string): Q.Promise<string> {
```

Comment:
I feel like this can be moved to separate type/class. Thoughts ?

Reply:
Since one of the functions from within the class references it, it may be a little harder to do, but not impossible to do so.  I can move the function into a separate class and reference it from within the azure-arm-common class.